docs/markdown/snippets/optionrefactor.md

```diff
@@ -0,0 +1,45 @@
+## Per project subproject options rewrite
+
+A common requirement when building large projects with many
+subprojects is to build some (or all) subprojects with project options
+that are different from the "main project". This has been sort of
+possible in a limited way but is now possible in a general way. These
+additions can be added, changed and removed at runtime using the
+command line or, in other words, without editing existing
+`meson.build` files.
+
+Since this release you can specify _augments_, which are basically
+per-project option settings. These can be specified for every top
+level (i.e. not project) options. Suppose you have a project that has
+a single subproject called `numbercruncher` that does heavy
+computation. During development you want to build that subproject with
+optimizations enabled but your main project without
+optimizations. This can be done by specifying an augment to the given
```

Comment:
The first tow times I read "augment" I thought it was "argument" misspelled. IIUC, augments are subproject-specific options. Can't the concept be called "subproject-specific option" instead of "augment"? I think it would be clearer and I don't think it need to be written so often for the lengthy name to be a problem.

Reply:
Originally I wanted to call it "override", but that is already used for per-target option values, so no. I guess we can call them "subproject specific options" in docs, but internally we need a more concise name. And in any case it would be nice to have the same name for this feature internally and externally.

 "Augment" was the best I could come up with, better ideas welcome.

Reply:
dumb thought: call it an augmentation instead of an augment and no one will get it confused with an argument?

Reply:
> "Augment" was the best I could come up with, better ideas welcome.

Supersede or Negate might be considered. But expanding them to "supersedes" or "Negates" or "Negations" fouls things up. Then there is [supersession](https://www.merriam-webster.com/dictionary/supersession), where the main advantage is that it's a long way from `argument'. ;)